core/src/main/java/org/elasticsearch/cluster/routing/allocation/RerouteExplanation.java

```diff
@@ -52,23 +52,21 @@ public Decision decisions() {
     }
 
     public static RerouteExplanation readFrom(StreamInput in) throws IOException {
-        String commandName = in.readString();
-        AllocationCommand command = AllocationCommands.lookupFactorySafe(commandName).readFrom(in);
+        AllocationCommand command = in.readAllocationCommand();
```

Comment:
This is a fair objection to our direction. I don't understand why we're using this pattern of StreamInput knowing about all the registered things. I'm just extending the pattern we've been using. I think I'd prefer that StreamInput/StreamOutput made readNamedWriteable/writeNamedWriteable public. That'd prevent it from having to know all of the categories.